### PR TITLE
Merge to one crate with bin targets

### PR DESCRIPTION
**Pull request by sornas**
This makes it easier to write something REPL-style which would probably be a better fit for windows. Maybe it makes a GUI/TUI easier too?

We actually get very comparable compilation times. Seems our crate structure didn't really benefit from crates to start with. This is all with `lto = "thin"` in release (and only once with lots of noise in general).

<table>
<tr>
	<td> Version
	<td> Clean (debug)
	<td> touch mumd and mumctl only (debug)
	<td> touch everything (debug)
	<td> Clean (release)
	<td> touch mumd and mumctl only (release)
	<td> touch everything (release)
<tr>
	<td> Split (main)
	<td> 38.52 s
	<td> 5.91 s
	<td> 16.47 s
	<td> 51.16 s
	<td> 17.83 s
	<td> 31.55 s
<tr>
	<td> Monocrate (this PR)
	<td> 35.86 s
	<td> 5.54 s
	<td> 18.27 s
	<td> 52.45 s
	<td> 18.55 s
	<td> 32.02 s
</table>

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,13 +1,53 @@
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+[package]
+name = "mum"
+version = "0.4.0"
```
Comment (default-username-852) [inline Cargo.toml:3]: I think this constitutes a major version bump, so we should bump this to 0.5.0.

**Review by default-username-852 — COMMENTED** [review]
This looks good in general, but I don't like that we integrate mumlib. I think a goal of the project is that other people should be able to create their own programs that can talk to mumd, and be able to use the definitions in mumlib to do so. With integrating mumlib into mum, this isn't possible anymore.

Also, the name `mum` is taken on crates.io, so maybe we want to call the crate `mumd` instead or something.

**Comment by default-username-852** [comment]
Also, we should update the other `mum*` crates to signal that we've restructured the project when these changes land.

**Comment by sornas** [comment]
#135

**Pull request closed without merging**